brokerapi/brokers/cloudsql/common-definition.go

```diff
@@ -17,167 +17,198 @@ package cloudsql
-	},
-	{
-		FieldName: "activation_policy",
-		Type:      broker.JsonTypeString,
-		Details:   "The activation policy specifies when the instance is activated; it is applicable only when the instance state is RUNNABLE.",
-		Default:   "ON_DEMAND",
-		Enum: map[interface{}]string{
-			"ALWAYS":    "Always, instance is always on.",
-			"NEVER":     "Never, instance does not turn on if a request arrives.",
-			"ON_DEMAND": "On Demand, instance responses to incoming requests and turns off when not in use.",
+func commonBindComputedVariables() []varcontext.DefaultVariable {
+	return []varcontext.DefaultVariable{
+		// legacy behavior dictates that empty values get defaults
+		{Name: "password", Default: `${password == "" ? rand.base64(32) : password}`, Overwrite: true},
+		{Name: "username", Default: `${username == "" ? "sb${str.truncate(14, time.nano())}" : username}`, Overwrite: true},
+	}
+}
+
+func commonProvisionVariables() []broker.BrokerVariable {
+	return []broker.BrokerVariable{
+		{
+			FieldName: "binlog",
+			Type:      broker.JsonTypeString,
+			Details:   "Whether binary log is enabled. If backup configuration is disabled, binary log must be disabled as well. Defaults: `false` for 1st gen, `true` for 2nd gen, set to `true` to use.",
+			Enum: map[interface{}]string{
+				"true":  "use binary log",
+				"false": "do not use binary log",
+			},
+		},
+		{
+			FieldName: "disk_size",
+			Type:      broker.JsonTypeString,
+			Details:   "In GB (only for 2nd generation instances).",
+			Default:   "10",
+			Constraints: validation.NewConstraintBuilder().
+				Pattern("^[1-9][0-9]+$").
+				MaxLength(5).
+				Examples("10", "500", "10230").
+				Build(),
+		},
+		{
+			FieldName: "region",
+			Type:      broker.JsonTypeString,
+			Details:   "The geographical region. See the instance locations list https://cloud.google.com/sql/docs/mysql/instance-locations for which regions support which databases.",
+			Default:   "us-central",
+			Constraints: validation.NewConstraintBuilder().
+				Pattern("^[A-Za-z][-a-z0-9A-Z]+$").
+				Examples("northamerica-northeast1", "southamerica-east1", "us-east1").
+				Build(),
+		},
+		{
+			FieldName: "zone",
+			Type:      broker.JsonTypeString,
+			Details:   "(only for 2nd generation instances)",
+			Constraints: validation.NewConstraintBuilder().
+				Pattern("^[A-Za-z][-a-z0-9A-Z]+$").
+				Build(),
+		},
+		{
+			FieldName: "disk_type",
+			Type:      broker.JsonTypeString,
+			Details:   "(only for 2nd generation instances)",
+			Default:   "PD_SSD",
+			Enum: map[interface{}]string{
+				"PD_SSD": "flash storage drive",
+				"PD_HDD": "magnetic hard drive",
+			},
+		},
+		{
+			FieldName: "maintenance_window_day",
+			Type:      broker.JsonTypeString,
+			Details:   "(only for 2nd generation instances) This specifies when a v2 CloudSQL instance should preferably be restarted for system maintenance puruposes. Day of week (1-7), starting on Monday.",
```

Comment:
s/puruposes/purposes

Reply:
Aww, man! That's a direct copy/paste from the docs. It looks like it rolls all the way back up through the client libs to the proto. Guess I'll be filing some bugs today.